projects/m2/025-checking-for-a-winning-card/js/index.js

```diff
@@ -0,0 +1,77 @@
+function generateBingoCard() {
+    let ranges = [[1, 15], [16, 30], [31, 45], [46, 60], [61, 75]];
+    let headers = ['B', 'I', 'N', 'G', 'O'];
+    let bingoColumns = {};
+
+    for (let i = 0; i < 5; i++) {
+        let column = [];
+        for (let j = 0; j < 5; j++) {
+            let min = ranges[i][0];
+            let max = ranges[i][1];
+            let randomNum = Math.floor(Math.random() * (max - min + 1)) + min;
+            column.push(randomNum);
+        }
+        bingoColumns[headers[i]] = column;
+    }
+
+    return bingoColumns;
+}
+
+function printBingoCard(bingoColumns) {
+    console.log(' B  I  N  G  O');
+    for (let i = 0; i < 5; i++) {
+        let row = '';
+        for (let header of ['B', 'I', 'N', 'G', 'O']) {
+            row += String(bingoColumns[header][i]).padStart(2, ' ') + ' ';
+        }
+        console.log(row);
+    }
+}
+
+function checkBingoCard(card) {
+    let headers = ['B', 'I', 'N', 'G', 'O'];
+
+    for (let i = 0; i < 5; i++) {
+        if (headers.every(header => card[header][i] === 0)) {
+            return true;
+        }
+    }
+
+    for (let header of headers) {
+        if (card[header].every(num => num === 0)) {
+            return true;
+        }
+    }
+
+    if (headers.every((header, i) => card[header][i] === 0)) {
+        return true;
+    }
+    if (headers.every((header, i) => card[header][4 - i] === 0)) {
+        return true;
+    }
+
+    return false;
+}
+
+function main() {
+   
+    let card1 = generateBingoCard();
+    let card2 = generateBingoCard();
+    let card3 = generateBingoCard();
+    let card4 = generateBingoCard();
```

Comment:
Pay attention that these are **variables** **containing** an **object**, therefore they contain the **reference** to the **memory** **address** and _not the simple value as happens for primitive data_ (string, number, etc.).
For this reason, you can use `const`, even if you modify it partially. ;-)